File: research/probes/scan_financial_chunks.mjs
// research/probes/scan_financial_chunks.mjs
async function main() {
  const url = 'https://storage.stockbit.com/financial/1.1.10/static/remoteEntry.js';
  const r = await fetch(url);
  const text = await r.text();

  const chunkMapMatch = text.match(/\{(?:\d+:"[a-f0-9]+",?)+\}/);
  if (!chunkMapMatch) return;

  const raw = chunkMapMatch[0];
  const fixed = raw.replace(/(\d+):/g, '"$1":');
  const chunkMap = JSON.parse(fixed);
  console.log(`Found ${Object.keys(chunkMap).length} chunks in financial MFE.`);

  const foundUrls = new Set();
  const ids = Object.keys(chunkMap);

  for (const chunkId of ids) {
    const chunkUrl = `https://storage.stockbit.com/financial/1.1.10/static/${chunkId}.${chunkMap[chunkId]}.js`;
    try {
      const cRes = await fetch(chunkUrl);
      const cText = await cRes.text();
      const apis = Array.from(cText.matchAll(/['"`](\/(?:findata|order-trade|company|emitten|keystats|market|stream|analyst|corpaction|insider|financial)[a-zA-Z0-9_\-/]*)['"`]/g)).map(m => m[1]);
      for (const a of apis) foundUrls.add(a);
    } catch(e) {}
  }

  console.log('\nAPI Endpoints found in financial MFE chunks:');
  for (const u of Array.from(foundUrls).sort()) {
    console.log(' ', u);
  }
}
main();

File: research/probes/scan_financial_chunks.py
import urllib.request
import json
import re

url = 'https://storage.stockbit.com/financial/1.1.10/static/remoteEntry.js'
req = urllib.request.urlopen(url)
text = req.read().decode('utf-8')

# Find chunk map
chunk_map_match = re.search(r'\{(?:\d+:"[a-f0-9]+",?)+\}', text)
if chunk_map_match:
    # Convert js object to valid json
    raw = chunk_map_match.group(0)
    fixed = re.sub(r'(\d+):', r'"\1":', raw)
    chunk_map = json.loads(fixed)
    print(f"Found {len(chunk_map)} chunks in financial MFE.")
    
    found_urls = set()
    for chunk_id, chunk_hash in chunk_map.items():
        chunk_url = f"https://storage.stockbit.com/financial/1.1.10/static/{chunk_id}.{chunk_hash}.js"
        try:
            c_text = urllib.request.urlopen(chunk_url).read().decode('utf-8')
            # Look for api endpoints
            apis = re.findall(r'[\'"`](/(?:findata|order-trade|company|emitten|keystats|market|stream|analyst|corpaction|insider)[a-zA-Z0-9_\-/]*)[\'"`]', c_text)
            for a in apis:
                found_urls.add(a)
        except Exception as e:
            pass
            
    print("\nAPI Endpoints found in financial MFE chunks:")
    for u in sorted(found_urls):
        print(" ", u)
